Architectural documentation (#12)
Energinet-DataHub/energy-origin: doc/api/datasync.md

# Data Sync Domain

The emissions api should take three query parameters

# Create relations

## Request

```text
POST /relations

{
    "ssn": null,
    "tin": 12345678
}
```

## Parameters

- ssn: Social security number
- tin: Company tin number

***ssn and tin are mutual exclusive, one must be set.***

## Response

```json
{
    "success": true
}
```

> ## Create relations for company

Reviewer: ```suggestion
## Internal call structure
```

```mermaid
sequenceDiagram
    autonumber
    participant spa as Single Page Application
    participant auth as Auth Domain
    participant datasync as DataSync Domain

>     spa->>+auth: login
>         auth ->>+ datasync: POST: /relation  {tin=12345678}
>         datasync -->>- auth: Success
>     auth-->>-spa: jwt / opaque token (cookie)

Reviewer: ```suggestion
    spa->>+auth: login
        auth ->>+ datasync: POST: /relations  {tin=12345678}
        datasync -->>- auth: Success
    auth-->>-spa: jwt / opaque token (cookie)
```

```

# Get Metering points

## Request

```text
GET /meteringpoints
```

## Parameters

\-

## Response

```json
{
    "meteringpoints": [
        {
            "meteringPointId": 57131300000000001
        }
    ]
}
```

# Get Measurements

## Request

```text
GET /measurements
    ?meteringpoint=123;456;789
    &dateFrom=1514826000
    &dateTo=1514864000
```

## Parameters

> - meteringpoint: 17 digit integer id of a meteringpoint, multiple can be seperated with ;

Reviewer 2: ```suggestion
- meteringpoint: 18 digit integer id(GSRN) of a meteringpoint, multiple can be seperated with ;
```
